app/models/title.rb

```diff
@@ -13,6 +13,39 @@ class Title < RmApiResource
         raise ActionController::BadRequest, 'Invalid filter parameter'
       end
 
+      querykeys = filters.keys & %w[name isxn subject publisher]
+
+      unless querykeys.size <= 1
+        raise ActionController::BadRequest, 'Conflicting filter parameters'
+      end
+
+      titlename = filters[:name]
+      isxn = filters[:isxn]
+      subject = filters[:subject]
+      publisher = filters[:publisher]
+      query = request.get_params.delete(:q)
+
+      if query && querykeys.size == 1
+        raise ActionController::BadRequest, 'Conflicting query parameters'
+      end
+
+      if query
+        request.get_params[:search] = query
+        request.get_params[:searchfield] ||= 'titlename'
```

Comment:
I don't think we need `||=` in these cases. This implies that we will allow a `?searchfield` param to supersede any of these rules.

Reply:
Thanks for catching that - made that adjustment